### PR TITLE
use busy stack instead of binary busy state

### PR DESCRIPTION
**Pull request by sjmiles**

### CODE REVIEW

File: pkg/ts/core/Particle.ts [hunk 1]
```diff
@@ -226,6 +226,7 @@ export class Particle {
     this.pipe = pipe;
     this.impl = create(proto);
     defineProperty(this, 'internal', privateProperty(nob()));
+    this.internal.$busy = 0;
```
Comment (shaper) [inline pkg/ts/core/Particle.ts:229]: What's a '$' prefix on a field in TS, is it still just the symbol name or does it have some effect?

**Review by shaper — APPROVED** [review]

File: pkg/ts/core/Particle.ts (hunk 1 above)
Comment (sjmiles) [reply pkg/ts/core/Particle.ts:229]: `$` is legal in an identifier, so there is no explicit effect. Usually we use it to denote a 'keyword' in a JSON structure, but here it's (subjectively) just a mistake. I think I intended this to mean `private`, but doesn't `internal` do that?

File: pkg/ts/isolation/ses.ts [hunk 2]
```diff
@@ -79,44 +79,57 @@ const requireImplFactory = async (kind, options) => {
...
+  // array up the function names
   const funcNames = funcs.map(([n]) => n);
-  //
+  // if it's not a Function, it's a const
   const consts = props.filter(item => !isFunc(item));
+  // build const decls
   const rewriteConsts = consts.map(([n, p]) => {
     return `const ${n} = \`${p}\`;`;
   });
+  // array up the const names
```
Comment (Cypher1) [inline pkg/ts/isolation/ses.ts:125]: `array of` perhaps?

File: pkg/ts/isolation/ses.ts [hunk 3]
```diff
@@ -79,44 +79,57 @@ const requireImplFactory = async (kind, options) => {
...
-${[...rewriteConsts, ...rewriteFuncs].join('\n\n')}
-
-return harden(${proto});
-
-//# sourceURL=Library/${pathForKind(kind).split('/').pop()}
-};
-  `;
-  log('rewritten:\n\n', rewrite);
-  return particleCompartment.evaluate(rewrite);
+  return {
```
Comment (Cypher1) [inline pkg/ts/isolation/ses.ts:127]: Is the kind and/or path no longer intended to be tracked?

**Comment by Cypher1** [comment]
LGTM with two questions that may be suggestions for changes depending on the answer :)

**Review by mariakleiner — APPROVED** [review]

File: pkg/ts/isolation/ses.ts (hunk 2 above)
Comment (sjmiles) [reply pkg/ts/isolation/ses.ts:125]: I'm referring to the action, not the object, but perhaps I was too informal.

**Comment by sjmiles** [comment]
We needed the fix for downstream apps, so I committed. @Cypher1: is not intended to gloss over your questions. Let's follow up.

**Comment by Cypher1** [comment]
> We needed the fix for downstream apps, so I committed. @Cypher1: is not intended to gloss over your questions. Let's follow up.

100%, I LGTM'd it.

File: pkg/ts/isolation/ses.ts (hunk 3 above)
Comment (Cypher1) [reply pkg/ts/isolation/ses.ts:127]: We used to add the path to the source and I think we don't anymore. Might be handy to have?

**Pull request merged**